assets/new-user-experience/modules/configure-sync/index.js

```diff
@@ -0,0 +1,304 @@
+/**
+ * External dependencies.
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import parse from 'html-react-parser';
+import {
+	Button,
+	SelectControl,
+	Modal,
+	CheckboxControl,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies.
+ */
+import {
+	Section,
+	SectionTitle,
+	SectionDescription,
+	InputWrapper,
+	SquareCheckboxControl,
+} from '../../components';
+import { useSquareSettings } from '../../settings/hooks';
+
+export const ConfigureSync = ( { indent = 0 } ) => {
+	const {
+		settings,
+		squareSettingsLoaded,
+		setSquareSettingData,
+	} = useSquareSettings();
+
+	const [ updateImport, setUpdateImport ] = useState( false );
+	const [ isOpen, setOpen ] = useState( false );
+	const [ isImporting, setIsImporting ] = useState( false );
+	const [ importDoneNotice, setImportDoneNotice ] = useState( '' );
+	const openModal = () => setOpen( true );
+	const closeModal = () => setOpen( false );
+
+	const {
+		system_of_record = 'disabled',
+		enable_inventory_sync = 'no',
+		override_product_images = 'no',
+		hide_missing_products = 'no',
+		sync_interval = '0.25',
+		is_connected = false,
+	} = settings;
+
+	const sync_interval_options = [
```

Comment:
I've not gotten further yet so maybe I'll find the answer to this later but if I recall correctly, we have a PHP filter right now allowing these options to be changed. Is that something that can still be done in this new experience?

Reply:
We do not have a filter for options but I can see a filter in PHP `wc_square_sync_interval` to filter the selected interval. We are not using that in this new react-based settings page. To do so we may need to create an endpoint to get the value from the PHP filter into the JS file.